### ISSUE TITLE
Transformer ("class-transformer") metadata cannot be inherited for "X" class

### ISSUE THREAD

**Issue by julianpoemp** [nestjs/swagger#2726]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I updated nestjs and other dependencies to the latest version and I get an error message as soon as I start serving the application:

```
ERROR [MappedTypes] Transformer ("class-transformer") metadata cannot be inherited for "AccountCreateRequestDto" class.

ERROR [MappedTypes] TypeError: Found non-callable @@iterator
```

previous, working installation:

```
├── @nestjs/cache-manager@2.1.0
├── @nestjs/common@10.2.5
├── @nestjs/config@3.1.1
├── @nestjs/core@10.2.5
├── @nestjs/jwt@10.1.1
├── @nestjs/mapped-types@2.0.2
├── @nestjs/passport@10.0.2
├── @nestjs/platform-express@10.2.5
├── @nestjs/schematics@10.0.2
├── @nestjs/serve-static@4.0.0
├── @nestjs/swagger@7.1.13
├── @nestjs/testing@10.2.5
├── @nestjs/throttler@4.2.1
├── @nestjs/typeorm@10.0.0
├── class-transformer@0.5.1
├── class-validator@0.14.0
```

current, failing installation:

```
├── @nestjs/cache-manager@2.1.1
├── @nestjs/common@10.2.10
├── @nestjs/config@3.1.1
├── @nestjs/core@10.2.10
├── @nestjs/jwt@10.2.0
├── @nestjs/mapped-types@2.0.4
├── @nestjs/passport@10.0.2
├── @nestjs/platform-express@10.2.10
├── @nestjs/schematics@10.0.3
├── @nestjs/serve-static@4.0.0
├── @nestjs/swagger@7.1.16  <-- error occurs with @nestjs/swagger >= 7.1.15
├── @nestjs/testing@10.2.10
├── @nestjs/throttler@5.0.1
├── @nestjs/typeorm@10.0.1
├── class-transformer@0.5.1
├── class-validator@0.14.0
```

AccountCreateRequestDto:

```typescript
export class AccountCreateRequestDto extends PartialType(OmitType(AccountRegisterRequestDto, ['redirectTo'])) {
...
}
```
I'm using NodeJS 18 and Nx 17.0.3

### Minimum reproduction code

https://github.com/julianpoemp/nestjs-swagger-issues

### Steps to reproduce

Unfortunately, I wasn't able to reproduce the error in a minimum reproduction code.

1. npm install
2. npm start

Now the error should appear, but it doesn't. The project where the error appears is not public, I can't reference it.

### Expected behavior

The error messages should disappear.

### Package version

7.1.16

### NestJS version

10.2.10

### Node.js version

18.18.2

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

The error disappears when I downgrade @nestjs/swagger to 7.1.14. The other dependencies can be updated to the latest version.

**Comment by kamilmysliwiec** [comment]
Might be related to https://github.com/nestjs/mapped-types/pull/1225#issue-1992458711 cc @KuanWenChen

**Comment by KuanWenChen** [comment]
@julianpoemp 
I can't  reproduce the problem.
![image](https://github.com/nestjs/swagger/assets/63909422/e7e58895-55a9-4bb2-b4fa-332e399106ba)

[This simple code ](https://github.com/KuanWenChen/mapped-types-issue/blob/master/index.ts) also can't reproduce the problem.

![image](https://github.com/nestjs/swagger/assets/63909422/59d81869-9433-4bbc-9cfe-30993720f605)

**Comment by KuanWenChen** [comment]
But I will try to rewrite the code to minimize the impact on the package

**Comment by julianpoemp** [comment]
@KuanWenChen yes, the repository I created to reproduce this issue doesn't trigger the error. I don't know why, but I know that this error occurs in my other (private) project and that it's solved when I downgrade @nestjs/swagger...

At the moment I don't know how to make this error reproducible. I'll trying...

**Comment by KuanWenChen** [comment]
I think I can't rollback something to mapped-type v2.0.2. It's going to cause _typeMetadatas, _exposeMetadatas, and _excludeMetadatas to inherit incorrectly ([Test Failed](https://github.com/nestjs/mapped-types/blob/8b29fb4143a6f7f8fbc681adade15dd4cce3f19b/tests/omit-type.helper.spec.ts#L68C5-L117C6)) . 

If you have any case that could reproduce the problem, I would try to fix it.

**Comment by YeritsyanTigran** [comment]
Hi @KuanWenChen ! I managed to reproduce this error in this [repository](https://github.com/YeritsyanTigran/nestjs-swagger-error-reproduce/tree/master/src) . In short the error is reproducible when you have something like this. 
```
@Exclude()
export class UserPublicDto {
    @Expose() @IsString()
    username: string;
}

@Exclude()
export class UserDto extends UserPublicDto {
    @Expose() @IsString()
    password: string;
}

@Exclude()
export class UserUpdateDto extends OmitType(UserDto, ['password']) {

}
```
The OmitType is important here as well as that we are using it on a class which already extends from another class. This issue is not reproducible on 7.1.14

**Comment by julianpoemp** [comment]
@YeritsyanTigran thank you very much for the reproduction!

**Comment by KuanWenChen** [comment]
@kamilmysliwiec 
This problem cannot be reproduced in @nestjs/mapped-types@2.0.4. I think it has been fixed in [mapped-types-issue-1225](https://github.com/nestjs/mapped-types/pull/1225#issue-1992458711). The package dependency has not been updated to 2.0.4, as indicated [here](https://github.com/nestjs/swagger/blob/f183ef776c7e3cc4c9b6cc6e1e34a89103d7b65d/package.json#L27C10-L27C10).

**Comment by julianpoemp** [comment]
Is it solved or closed because it couldn't be reproduced?

**Comment by kamilmysliwiec** [comment]
It should be fixed in 7.1.17
